chpl/chpl-service/src/main/java/gov/healthit/chpl/app/SurveillanceOversightReportWeeklyApp.java

```diff
@@ -1,115 +1,99 @@
-		 app.setCpdManager((CertifiedProductDetailsManager)context.getBean("certifiedProductDetailsManager"));
-		 app.setCertifiedProductDAO((CertifiedProductDAO)context.getBean("certifiedProductDAO"));
-		 app.setPresenter((SurveillanceOversightAllBrokenRulesCsvPresenter)context.getBean("surveillanceOversightAllBrokenRulesCsvPresenter"));
-		 app.getPresenter().setProps(props);
-		 app.setMailUtils((SendMailUtil)context.getBean("SendMailUtil"));
-		 
-		 //where to store this file
-        String downloadFolderPath;
-        if (args.length > 0) {
-        	downloadFolderPath = args[0];
-        } else {
-        	downloadFolderPath = props.getProperty("downloadFolderPath");
-        }
-        File downloadFolder = new File(downloadFolderPath);
-        if(!downloadFolder.exists()) {
-        	downloadFolder.mkdirs();
-        }
         
-        List<CertifiedProductDetailsDTO> allCertifiedProducts = app.getCertifiedProductDAO().findWithSurveillance();
-        List<CertifiedProductSearchDetails> allCertifiedProductDetails = new ArrayList<CertifiedProductSearchDetails>(allCertifiedProducts.size());
-		for(CertifiedProductDetailsDTO currProduct : allCertifiedProducts) {
-			try {
-				CertifiedProductSearchDetails product = app.getCpdManager().getCertifiedProductDetails(currProduct.getId());
-				allCertifiedProductDetails.add(product);
-			} catch(EntityRetrievalException ex) {
-				logger.error("Could not certified product details for certified product " + currProduct.getId());
-			}
-		}
-		CertifiedProductDownloadResponse allCps = new CertifiedProductDownloadResponse();
-		allCps.setProducts(allCertifiedProductDetails);
+        context.close();
+	}
+	
+	protected void initiateSpringBeans(AbstractApplicationContext context, Properties props){
+		super.initiateSpringBeans(context, props);
+		this.setPresenter((SurveillanceOversightAllBrokenRulesCsvPresenter)context.getBean("surveillanceOversightAllBrokenRulesCsvPresenter"));
+		this.getPresenter().setProps(props);
+	}
+	
+	private void sendOncWeeklyEmail(Set<String> oncWeeklyEmails, Properties props, File downloadFolder, CertifiedProductDownloadResponse cpList) throws IOException, AddressException, MessagingException {
+		String surveillanceReportFilename = null;
+        String htmlMessage = null;
+        String subject = null;
+        File surveillanceReportFile = null;
 		
-        //write out a csv file containing all surveillance
-        String surveillanceReportFilename = downloadFolder.getAbsolutePath() + File.separator + FILENAME;
-        File surveillanceReportFile = new File(surveillanceReportFilename);
+		surveillanceReportFilename = props.getProperty("oversightEmailWeeklyFileName");
+    	surveillanceReportFile = new File(downloadFolder.getAbsolutePath() + File.separator + surveillanceReportFilename);
         if(!surveillanceReportFile.exists()) {
         	surveillanceReportFile.createNewFile();
         } else {
         	surveillanceReportFile.delete();
```

Comment:
I think I wrote this bit of code so I'll try to answer... and it may be that the code isn't clear enough/right (although it's working). I think the intent is to either create a new file or erase the contents of an existing file since we are creating files with the same names every time.  I was trying to just always make sure we started writing in a blank file rather than possibly have any of the content in the file last time it was written get mixed in with the current write. Dan, do you want to see if there is a better way to do this? 

The file gets immediately re-created when presentAsFile() is called in the next few lines. It's always written before it's attached to any emails.